R/plotLoadings.R

```diff
@@ -0,0 +1,341 @@
+            # Ordering loadings and adding factor to keep the order
+            L <- .get_loadings_plot_data(loadings_matrix, n, ncomponents)
+        } else {
+            stop("These methods are yet to be implemented", call. = FALSE)
+        }
+        
+        if (layout == "tree") {
+            # Plot tree with feature loadings
+            p <- .loadings_tree_plotter(x, loadings_matrix, ncomponents)
+        } else {
+            # Plot features with the layout selected
+            p <- .plot_pca_feature_loadings(L, layout, ncomponents)
+        }
+    return(p)
+    }
+)
+#' @rdname plotLoadings
+#' @export 
+setMethod("plotLoadings", signature = c(x = "matrix"),
+    function(x,
+            dimred = "PCA",
+            layout = "heatmap",
+            n = 10,
+            name = NULL,
+            ...) {
+        # Making sure there is no error in parameters given by the user
+        .check_parameters(x,
+                        dimred = dimred,
+                        layout = layout,
+                        n = n,
+                        ...)
+                      
+        if (dimred == "PCA") {
+            # Set number of components
+            ncomponents <- length(colnames(x))
+            # Ordering loadings and adding factor to keep the order
+            df <- .get_loadings_plot_data(x, n, ncomponents)
+            # Plot features with the layout selected
+            p <- .plot_pca_feature_loadings(df, layout, ncomponents)
+        } else {
+            stop("These methods are yet to be implemented", call. = FALSE)
+        }
+    return(p)
+    }
+)
+
+.check_parameters <- function(x, dimred, layout, n,...) {
+    # Checking if dimred is correct
+    if ( !(dimred %in% c("PCA", "LDA", "NMF")) ) {
+        stop("'dimred' must be one of c('PCA', 'LDA', 'NMF').", call. = FALSE)
+    }
+    # Checking if layout is correct
+    if ( !(layout %in% c("screeplot", "barplot", "tree", "heatmap")) ) {
+        stop("'dimred' must be one of c('screeplot', 'barplot', 'tree', 'heatmap').", call. = FALSE)
+    }
+    # Making sure the user doesn't try to plot the tree if he gives only the matrix
+    if (is.matrix(x) && layout == "tree") {
+        stop("TreeSummarizedExperiment object is required for the tree plotting.", call. = FALSE)
+    }
+    # Making sure the tree is not null
+    if (is(x, "TreeSummarizedExperiment")) {
+        if ( is.null(rowTree(x)) && layout == "tree") {
+            stop ("Tree is null.", call. = FALSE)
+        }
+    }
+    #Checking if n is a positive number
+    if ( !(is.numeric(n) && n > 0) ) {
+        stop("'n' must be a positive number.", call. = FALSE)
+    }
+
+}
+
+.get_loadings_plot_data <- function(x, n, ncomponents) {
+    L <- list()
+    for (i in 1:ncomponents) {
+        df <- data.frame(x)
+        df[["Feature"]] <- rownames(x)
+        # Ordering loadings
+        df <- df[ order(abs(df[[paste("PC",as.character(i),sep="")]])[1:n], decreasing = TRUE), ]
```

Comment:
I believe you can do this without loop. (Loops should be avoided in R if possible, they are slow. Also good practice to avoid loops)

Moreover, you are handling the data with column names. However, it is not necessary since you know the index of columns. It is preferred to use indices since that is more general approach (column names can differ; they are not always "PC*")